gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/domain_service/ApiMetadataDomainService.java

```diff
@@ -86,21 +84,31 @@ public void createDefaultApiMetadata(String apiId, AuditInfo auditInfo) {
 
     public void saveApiMetadata(String apiId, List<ApiMetadata> metadata, AuditInfo auditInfo) {
         var previousMetadata = apiMetadataQueryService.findApiMetadata(apiId);
+        updateApiMetadata(apiId, metadata, auditInfo, previousMetadata.keySet());
+    }
 
-        if (metadata != null) {
-            for (var metadataEntry : metadata) {
+    public void updateApiMetadata(String apiId, List<ApiMetadata> metadata, AuditInfo auditInfo) {
```

Comment:
I'm not convinced by the naming of the public methods. I have a suggestion: what about 
- renaming the existing `saveApiMetadata` into `importApiMetadata`. We can add a Javadoc explaining that it will remove metadata that exists but not anymore in the list provided
- renaming the new `updateApiMetadata` into `saveApiMetadata` with a Javadoc explaining it will only process the list provided

I don't like `updateApiMetadata` because under the hood we can create items... We can eventually use `replaceApiMetadata` instead of `importApiMetadata`. The good side of `import...` is that we could now it is used by "import" use cases 🤔